robotiq_modbus_rtu/src/robotiq_modbus_rtu/comModbusRtu.py

```diff
@@ -33,41 +33,48 @@
 # Copyright (c) 2012, Robotiq, Inc.
 # Revision $Id$
 #
-# Modifed from the orginal comModbusTcp by Kelsey Hawkins @ Georgia Tech
+# Modifed from the original comModbusTcp by Kelsey Hawkins @ Georgia Tech
 
 
-
-"""@package docstring
-Module comModbusRtu: defines a class which communicates with Robotiq Grippers using the Modbus RTU protocol. 
-
-The module depends on pymodbus (http://code.google.com/p/pymodbus/) for the Modbus RTU client.
-"""
-
-from pymodbus.client.sync import ModbusSerialClient
+import fcntl
+import ready_logging
 from math import ceil
+from threading import Lock
+from pymodbus.client.sync import ModbusSerialClient
+
+ROBOTIQ_ID = 0x09
 
 
-class communication:
+@ready_logging.logged
+class Communication(object):
     def __init__(self):
         self.client = None
+        self.modbus_action_lock = Lock()
 
-    def connectToDevice(self, device):
-        """Connection to the client - the method takes the IP address (as a string, e.g. '192.168.1.11') as an argument."""
+    def connect_to_device(self, device):
+        """ Connect to the client """
         self.client = ModbusSerialClient(method='rtu', port=device, stopbits=1, bytesize=8, baudrate=115200,
                                          timeout=0.2)
+
         if not self.client.connect():
-            print "Unable to connect to %s" % device
+            self.__log.err('Unable to connect to {}'.format(device))
             return False
+        fcntl.flock(self.client.socket, fcntl.LOCK_EX)
         return True
 
-    def disconnectFromDevice(self):
+    def disconnect_from_device(self):
         """Close connection"""
+        fcntl.flock(self.client.socket, fcntl.LOCK_UN)
         self.client.close()
 
-    def sendCommand(self, data):
-        """Send a command to the Gripper - the method takes a list of uint8 as an argument. The meaning of each variable depends on the Gripper model (see support.robotiq.com for more details)"""
+    def send_command(self, data):
+        """
+            Send a command to the Gripper - the method takes a list of uint8 as an argument. The meaning of each
+            variable depends on the Gripper model (see support.robotiq.com for more details)
+        """
+
         # make sure data has an even number of elements
-        if (len(data) % 2 == 1):
+        if len(data) % 2 == 1:
```

Comment:
I don't think this comment is correct.

len(data) % 2 == 0 when data is even.

Reply:
That is leftover from Robotiq. They want an even number, since it is 2 bytes sent per register